palace/models/materialoperator.cpp

```diff
@@ -461,14 +456,16 @@ void MaterialOperator::SetUpMaterialProperties(const IoData &iodata,
         std::abs(data.lambda_L) > 0.0 ? std::pow(data.lambda_L, -2.0) : 0.0;
     if (mat_invLondon(count).MaxMaxNorm() > 0.0)
     {
-      for (auto attr : data.attributes)
-      {
-        london_attr.Append(attr);
-      }
+      has_london_attr = true;
     }
 
     count++;
   }
+  int has_attr[3] = {has_losstan_attr, has_conductivity_attr, has_london_attr};
+  Mpi::GlobalMax(3, has_attr, mesh.GetComm());
+  has_losstan_attr = has_attr[0];
+  has_conductivity_attr = has_attr[1];
+  has_london_attr = has_attr[2];
```

Comment:
Was wondering about why the need for `int` register, and see it doesn't work with `bool` directly. It seems more elegant to have
```
  static void GlobalOr(int len, bool *buff, MPI_Comm comm)
  {
    GlobalOp(len, buff, MPI_LOR, comm);
  }

  static void GlobalAnd(int len, bool *buff, MPI_Comm comm)
  {
    GlobalOp(len, buff, MPI_LAND, comm);
  }
```
in `communicator.hpp` and use it directly. Could also have specializations of `Max` and `Min` for `bool` if really wanted, but I think `Or` and `And` are semantically clearer.

Second to that, can then have `struct {bool losstan, conductivity, london;} has_attr` and send the struct directly with `GlobalOr(3, &has_attr.losstan, mesh.GetComm());`

Reply:
Resolved in [31329d9](https://github.com/awslabs/palace/pull/201/commits/31329d9cde4d9c105ad53f415ac0bdd03ab0a40e). Not going to go so far as to add a nested `has_attr` struct at this point from a style perspective.